VideoLocker/src/main/java/org/edx/mobile/social/google/GoogleOauth2.java

```diff
@@ -227,7 +235,10 @@ public void login() {
     public void logout() {
         if (accessToken != null) {
             try {
-                GoogleAuthUtil.clearToken(activity.get(), accessToken);
+                Activity activity = getActivity();
+                if ( activity == null )
+                    return;
+                GoogleAuthUtil.clearToken(activity, accessToken);
                 logger.debug("Google logged out");
             } catch (GooglePlayServicesAvailabilityException e) {
```

Comment:
@BenjiLee 
agree.  we should separate logic into two method :  one to get activity,  one to check if activity is healthy